journal/src/main/java/io/camunda/zeebe/journal/file/SegmentLoader.java

```diff
@@ -151,27 +152,11 @@ private SegmentDescriptor readDescriptor(final Path file) {
 
     try (final FileChannel channel = FileChannel.open(file, StandardOpenOption.READ)) {
       final var fileSize = Files.size(file);
-      final byte version = readVersion(channel, fileName);
-      final int length = SegmentDescriptor.getEncodingLengthForVersion(version);
-      if (fileSize < length) {
-        throw new CorruptedJournalException(
-            String.format(
-                "Expected segment '%s' with version %d to be at least %d bytes long but it only has %d.",
-                fileName, version, length, fileSize));
-      }
-
-      final ByteBuffer buffer = ByteBuffer.allocate(length);
-      final int readBytes = channel.read(buffer, 0);
-
-      if (readBytes != -1 && readBytes < length) {
-        throw new JournalException(
-            String.format(
-                "Expected to read %d bytes of segment '%s' with %d version but only read %d bytes.",
-                length, fileName, version, readBytes));
-      }
-
-      buffer.flip();
-      return new SegmentDescriptor(buffer);
+      // We don't know the length of segment descriptor, so we let
+      final var buffer = channel.map(MapMode.READ_ONLY, 0, fileSize);
+      final var descriptor = new SegmentDescriptorReader().readFrom(buffer);
+      IoUtil.unmap(buffer);
```

Comment:
:upside_down_face: The overhead of mapping/unmapping just to read the descriptor seems unnecessary. As it's not really in the hot path it's probably fine, but would be nice if we got rid of that. Maybe we just map it once initially for the length of the file, then only remap if the descriptor says the segment is actually longer (though unlikely, since we pre-allocate _before_ we write the descriptor)?

Reply:
> Maybe we just map it once initially for the length of the file

Do you mean to re-use the mapping for the Segment?

Reply:
Yeah, pretty much, so we only map it once in most cases. It's very minor though, can be skipped if it turns out it's a lot of refactoring